classes/boardmanager.php

```diff
@@ -0,0 +1,1152 @@
+
+    /** @var cm_info Course module info */
+    private cm_info $cminfo;
+
+    /** @var stdClass Course */
+    private stdClass $course;
+
+    /**
+     * Constructor
+     *
+     * @param int $cmid Course module id (0 if course module is not created yet)
+     * @param int $boardid Board id (if 0, no board is loaded at this time)
+     */
+    public function __construct(int $cmid = 0, int $boardid = 0) {
+        $this->cmid = $cmid;
+        if ($cmid) {
+            list($this->course, $this->cminfo) = get_course_and_cm_from_cmid($cmid);
+            $this->load_instance($this->cminfo->instance);
+        }
+        $this->formatter = new updateformatter();
+        if (!empty($boardid)) {
+            $this->load_board($boardid);
+        }
+    }
+
+    /**
+     * Load a kanban instance
+     *
+     * @param int $instance Instance id
+     * @param bool $dontloadcm Don't load course module data - only needed at instance creation time
+     * @return void
+     */
+    public function load_instance(int $instance, bool $dontloadcm = false): void {
+        global $DB;
+        $this->kanban = $DB->get_record('kanban', ['id' => $instance], '*', MUST_EXIST);
+        if (!$dontloadcm) {
+            list ($this->course, $this->cminfo) = get_course_and_cm_from_instance($this->kanban->id, 'kanban');
+            $this->cmid = $this->cminfo->id;
+        }
+    }
+
+    /**
+     * Load a board.
+     *
+     * @param int $id Id of the board
+     * @return void
+     */
+    public function load_board(int $id): void {
+        $this->board = helper::get_cached_board($id);
+        if (empty($this->cminfo)) {
+            $this->load_instance($this->board->kanban_instance);
+        }
+    }
+
+    /**
+     * Get the current board record.
+     *
+     * @return stdClass The current board
+     */
+    public function get_board(): stdClass {
+        return $this->board;
+    }
+
+    /**
+     * Return representation of collected updates.
+     *
+     * @return string
+     */
+    public function get_formatted_updates(): string {
+        return $this->formatter->get_formatted_updates();
+    }
+
+    /**
+     * Get the current template for this board. If there are multiple templates, use the latest one.
+     *
+     * @return int Board id of the template board, 0 if none found.
+     */
+    public function get_template_board_id(): int {
+        global $DB;
+        $result = $DB->get_records(
+            'kanban_board',
+            ['kanban_instance' => $this->kanban->id, 'template' => 1],
+            'timemodified DESC',
+            'id',
+            0,
+            1
+        );
+        if (!$result) {
+            // Is there a system-wide template?
+            $result = $DB->get_records('kanban_board', ['kanban_instance' => 0, 'template' => 1], 'timemodified DESC', 'id', 0, 1);
+        }
+        if (!$result) {
+            return 0;
+        } else {
+            return array_pop($result)->id;
+        }
```

Comment:
suggestion:

`
return $result ? array_pop($result)->id : 0;
`
Or, if you prefer the early exit syntax style:
```
if (!$result) {
  return 0;
}
return array_pop($result)->id;
```